app/lib/dartdoc/backend.dart

```diff
@@ -86,6 +86,23 @@ class DartdocBackend {
     return new PubDartdocData.fromJson(map);
   }
 
+  /// Deletes the old entries that predate [shared_versions.gcBeforeRuntimeVersion].
+  Future deleteOldSdkData() async {
+    final prefix = storage_path.dartSdkDartdocPrefix();
+    await for (BucketEntry entry in _storage.list(prefix: '$prefix/')) {
+      if (entry.isDirectory) {
+        continue;
+      }
+      final name = p.basename(entry.name);
+      final version = name.replaceAll('.json.gz', '');
+      if (version.length == 10 &&
+          version.compareTo('2018.00.00') > 0 &&
+          version.compareTo(shared_versions.gcBeforeRuntimeVersion) < 0) {
+        await _storage.delete(entry.name);
```

Comment:
This prevents us from easily reverting back beyond gcBeforeRuntimeVersion as the newer version will interfere with the older version's data, which is none of its business.

Is the .json.gz updated continuously by the instances using it? If so, we can check whether it has been updated in the past X days and then keep it. Otherwise, we could store a marker every now and then to signal that this particular older version is currently running, and other versions shouldn't touch its files.